qiskit/transpiler/preset_passmanagers/level0.py

```diff
@@ -157,14 +157,19 @@ def _swap_condition(property_set):
     elif translation_method == "translator":
         from qiskit.circuit.equivalence_library import SessionEquivalenceLibrary as sel
 
-        _unroll = [
-            UnitarySynthesis(
-                basis_gates,
-                approximation_degree=approximation_degree,
-                coupling_map=coupling_map,
-                backend_props=backend_properties,
-                method=unitary_synthesis_method,
-            ),
+        if basis_gates is not None and "unitary" not in basis_gates:
```

Comment:
I think this changes the default behaviour if `basis_gates` is `None` - this is the relevant part of `UnitarySynthesis.run` from immediately before #6124:  https://github.com/Qiskit/qiskit-terra/blob/c202e460a7369fde41209c2f41ad1484a3810a31/qiskit/transpiler/passes/synthesis/unitary_synthesis.py#L158-L160

In there, `if basis_gates is None` then the pass will run and synthesis unitary gates.   I would think the old form is _possibly_ what people would expect, but it's a bit of a funny one.  If we want to maintain the exact behaviour, then it needs to be `if (not basis_gates) or "unitary" not in basis_gates` (which also runs it if `basis_gates=[]`), or perhaps more sensibly, `if basis_gates is None or "unitary" not in basis_gates`.

Similar comments on every pass level.

Reply:
So, I've been testing this with 0.18.3 to compare the results with this PR:
```python3
qc = QuantumCircuit(2)
qc.unitary(random_unitary(4, seed=42), [0, 1])
qc.measure_all()
```

With `result = transpile(qc, basis_gates=None)`:

0.18.3:

```
        ┌──────────┐ ░ ┌─┐   
   q_0: ┤0         ├─░─┤M├───
        │  Unitary │ ░ └╥┘┌─┐
   q_1: ┤1         ├─░──╫─┤M├
        └──────────┘ ░  ║ └╥┘
meas: 2/════════════════╩══╩═
                        0  1 
```

This PR:
```
        ┌──────────┐ ░ ┌─┐   
   q_0: ┤0         ├─░─┤M├───
        │  Unitary │ ░ └╥┘┌─┐
   q_1: ┤1         ├─░──╫─┤M├
        └──────────┘ ░  ║ └╥┘
meas: 2/════════════════╩══╩═
                        0  1 
```

With `result = transpile(qc, basis_gates=[])`:

0.18.3:
```
qiskit.transpiler.exceptions.TranspilerError: "Unable to map source basis {('barrier', 2), ('u3', 1), ('measure', 1), ('cx', 2)} to target basis {'delay', 'reset', 'snapshot', 'barrier', 'measure'} over library <qiskit.circuit.equivalence.EquivalenceLibrary object at 0x7fbe564d2460>."
```

This PR:
```
qiskit.transpiler.exceptions.TranspilerError: "Unable to map source basis {('u3', 1), ('barrier', 2), ('measure', 1), ('cx', 2)} to target basis {'barrier', 'reset', 'measure', 'snapshot', 'delay'} over library <qiskit.circuit.equivalence.EquivalenceLibrary object at 0x7f2dcff347f0>."
```

So I think the translator method is ok as is. Unless I'm missing something or I need a different example circuit.

Reply:
For the record - the lines I highlighted above aren't the only relevant lines in 0.18.3.  For one, the `UnitarySynthesis` plugin wasn't even run at `optimization_level=1` in 0.18.3, because that didn't come til #6349, and for two, even if you force it, there's an implicit skip because `_choose_kak_gate(None)` would return `None`, and cause no synthesis to be done.